poc/dynamic-renderer.php

```diff
@@ -13,14 +13,13 @@ public function registerContentType($type, Callable $handler) {
 	public function __call($method, $arguments) {
 		if (preg_match('~^visit(.+)~', $method, $match)) {
 			$type = strtolower($match[1]);
-			if (!isset($this->handlers[$type])) {
-				$type = 'default';
-			}
```

Comment:
This `if` statement has no `else`. It just sets the type to 'default', if no specific handler is found, so the next `if` statement has a chance to pick it up instead.